src/core/memory.cc

```diff
@@ -1926,14 +1926,14 @@ void* realloc(void* ptr, size_t size) {
         // If ptr is a null pointer, the behavior is the same as calling std::malloc(new_size).
         return malloc(size);
     } else if (!is_seastar_memory(ptr)) {
-        // we can't realloc foreign memory on a shard
-        if (is_reactor_thread) {
-            abort();
-        }
         // original_realloc_func might be null when previous ctor allocates
         if (original_realloc_func) {
             return original_realloc_func(ptr, size);
         }
+        // we can't realloc foreign memory on a shard
```

Comment:
This comment is no longer correct. We reach here only in some esoteric case where original_realloc_func is missing, it's no longer a case that we can't realloc foreign memory.

Reply:
Yes